seatunnel-connectors-v2/connector-clickhouse/src/main/java/org/apache/seatunnel/connectors/seatunnel/clickhouse/config/ClickhouseSinkOptions.java

```diff
@@ -0,0 +1,107 @@
+package org.apache.seatunnel.connectors.seatunnel.clickhouse.config;
+
+import org.apache.seatunnel.api.configuration.Option;
+import org.apache.seatunnel.api.configuration.Options;
+import org.apache.seatunnel.api.sink.DataSaveMode;
+import org.apache.seatunnel.api.sink.SaveModePlaceHolder;
+import org.apache.seatunnel.api.sink.SchemaSaveMode;
+
+public class ClickhouseSinkOptions {
+
+    /** Bulk size of clickhouse jdbc */
+    public static final Option<Integer> BULK_SIZE =
+            Options.key("bulk_size")
+                    .intType()
+                    .defaultValue(20000)
+                    .withDescription("Bulk size of clickhouse jdbc");
+
+    /** Clickhouse table name */
+    public static final Option<String> TABLE =
+            Options.key("table")
+                    .stringType()
+                    .noDefaultValue()
+                    .withDescription("Clickhouse table name");
+
+    /** Split mode when table is distributed engine */
+    public static final Option<Boolean> SPLIT_MODE =
+            Options.key("split_mode")
+                    .booleanType()
+                    .defaultValue(false)
+                    .withDescription("Split mode when table is distributed engine");
+
+    /** When split_mode is true, the sharding_key use for split */
+    public static final Option<String> SHARDING_KEY =
+            Options.key("sharding_key")
+                    .stringType()
+                    .noDefaultValue()
+                    .withDescription("When split_mode is true, the sharding_key use for split");
+
+    public static final Option<String> PRIMARY_KEY =
+            Options.key("primary_key")
+                    .stringType()
+                    .noDefaultValue()
+                    .withDescription(
+                            "Mark the primary key column from clickhouse table, and based on primary key execute INSERT/UPDATE/DELETE to clickhouse table");
+
+    public static final Option<Boolean> SUPPORT_UPSERT =
+            Options.key("support_upsert")
+                    .booleanType()
+                    .defaultValue(false)
+                    .withDescription("Support upsert row by query primary key");
+
+    public static final Option<Boolean> ALLOW_EXPERIMENTAL_LIGHTWEIGHT_DELETE =
+            Options.key("allow_experimental_lightweight_delete")
+                    .booleanType()
+                    .defaultValue(false)
+                    .withDescription(
+                            "Allow experimental lightweight delete based on `*MergeTree` table engine");
+
+    public static final Option<SchemaSaveMode> SCHEMA_SAVE_MODE =
+            Options.key("schema_save_mode")
```

Comment:
Please also add this option parameter to the `Factory.optionRule()` method.
Every parameter in the `Options` class should be included in this method, with a description indicating whether it is optional or required.